modules/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/Init.java

```diff
@@ -53,8 +76,130 @@ public ConnectorFuture execute(Context context) {
             url = url.substring(0, url.length() - 1);
             connector.setStringField(0, url);
         }
+
+        HttpConnectionManager connectionManager = HttpConnectionManager.getInstance();
+
+        String scheme;
+        if (url.startsWith("http://")) {
+            scheme = Constants.PROTOCOL_HTTP;
+        } else if (url.startsWith("https://")) {
+            scheme = Constants.PROTOCOL_HTTPS;
+        } else {
+            throw new BallerinaException("Malformed URL: " + url);
+        }
+
+        Map<String, Object> properties =
+                HTTPConnectorUtil.getTransportProperties(connectionManager.getTransportConfig());
+        SenderConfiguration senderConfiguration =
+                HTTPConnectorUtil.getSenderConfiguration(connectionManager.getTransportConfig(), scheme);
+
+        if (connectionManager.isHTTPTraceLoggerEnabled()) {
+            senderConfiguration.setHttpTraceLogEnabled(true);
+        }
+        senderConfiguration.setTlsStoreType(Constants.PKCS_STORE_TYPE);
```

Comment:
setTLSStoreType

Reply:
This will have to come from the transport. Will change once a transport release goes on, with the API change.